spring-kafka/src/test/java/org/springframework/kafka/core/KafkaTemplateTests.java

```diff
@@ -195,4 +195,14 @@ public void testTemplateDisambiguation() throws Exception {
 		pf.createProducer().close();
 	}
 
+	@Test(expected = IllegalStateException.class)
+	public void flushWithoutSend() throws Exception {
+		Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+		DefaultKafkaProducerFactory<String, String> pf = new DefaultKafkaProducerFactory<String, String>(senderProps);
+		pf.setKeySerializer(new StringSerializer());
+		KafkaTemplate<String, String> template = new KafkaTemplate<>(pf);
+		template.setDefaultTopic(STRING_KEY_TOPIC);
+		template.flush();
+		throw new AssertionError("IllegalStateException must be thrown upon flushing");
```

Comment:
Hmm... Good idea about exception's fragment.
I also prefer the approach with `instanceOf` - had some floating issues when `expected` did not work as expected. However I've chosen it here as it's "junit-native" - thought it's more suitable for the framework. But was not sure about checking the whole exception message - did not want to force updating the tests in case you decide to do some reorganisation in texts. But the point about identifying the exact needed exception is good.

And sorry for missing `@author` update, most probably it's really time to sleep for me :) Have a good day!